lightning/src/ln/peer_handler.rs

```diff
@@ -174,6 +195,13 @@ pub struct MessageHandler<CM: Deref, RM: Deref> where
 	pub route_handler: RM,
 }
 
+/// Handler for messages external to the LN protocol.
+pub trait CustomMessageHandler : wire::Read {
```

Comment:
I'm not a huge fan of this being defined in a different file from `wire::Read`, tbh. Maybe it makes the most sense to put both here?

Reply:
Moved, but note that the reason for having them separated was to avoid circular dependency between wire and peer_handler as noted before (https://github.com/rust-bitcoin/rust-lightning/pull/1031#discussion_r681140736). Now it's back because `CustomMessageHandler` refers to `MessageHandlingError` which is in peer_handler.

Reply:
Ah, I mean I don't care either way, then. Feels strange, but whatever.

Reply:
I don't care either, @jkczyz do you prefer that I put it back in peer_handler to avoid the circular dependency?

Reply:
I'd prefer to keep them separated, but note that they can be independent traits. No need to make one a super trait of the other since you already constrain the generic type by both traits.

Reply:
I moved back the `CustomMessageHandler` trait into peer_handler.rs to remove the circular dependency. However I didn't manage to make them independent traits. The issue AFAIU is that the compiler needs to know that both traits' associated type are the same in order to be able to use the output of `read` as an input to `handle_custom_message`. Maybe it's possible to put this constraint somewhere else? (Although I must say even if it's possible I personally find it cleaner to have the constraint directly on the traits since I can't see a case where you'd want the types to differ).